examples/digital-identity/views/pages/share.ejs

```diff
@@ -0,0 +1,173 @@
+      let currentSessionId
+
+      const sessionIdResolver = async (policyType) => {
+        completionContainerDom.classList.add('hidden')
+        profileLinksDom.classList.add('hidden')
+
+        const urlSearchParamsPolicyType = policyType ? new URLSearchParams([ [ 'policyType', policyType ] ]) : new URLSearchParams()
+
+        const sessionId = await (await fetch(GET_NEW_SESSION_ID_URL+'?'+urlSearchParamsPolicyType)).text()
+        console.info('sessionIdResolver found id', sessionId);
+        currentSessionId = sessionId
+        return sessionId
+      };
+
+      const completionHandler = async (receivedReceiptId) => {
+        completionContainerDom.classList.remove('hidden')
+
+        console.log('Session completed - received receiptId:', receivedReceiptId);
+        const urlSearchParamsReceiptId = receivedReceiptId ? new URLSearchParams([ [ 'receiptId', receivedReceiptId ] ]) : new URLSearchParams()
+        const urlSearchParamsSessionId = currentSessionId ? new URLSearchParams([ [ 'sessionId', currentSessionId ] ]) : new URLSearchParams()
+
+        receiptDataContainerDom.innerText = 'Loading receipt data....'
+        try {
+          const response = await fetch(`${GET_RECEIPT_URL}?${urlSearchParamsReceiptId}`)
+          const contentTypeIsJSON = response.headers.get('Content-Type').includes('json')
+          let receiptData
+          if (response.ok) {
+            receiptData = await response.json()
+            profileLinksDom.classList.remove('hidden')
+            profileLinkBySessionIdDom.href = `${PROFILE_URL}?${urlSearchParamsSessionId}`
+            profileLinkByReceiptIdDom.href = `${PROFILE_URL}?${urlSearchParamsReceiptId}`
+          } else {
+            receiptData = contentTypeIsJSON ? await response.json(): `Error: ${await response.text()}`
+          }
+          receiptDataContainerDom.innerText = contentTypeIsJSON ? JSON.stringify(receiptData, null, 2) : receiptData
+        } catch (e) {
+          receiptDataContainerDom.innerText = `An error occurred when loading the receipt data: ${e.message}`
+        }
+      };
+
+      const errorListener = (errorCode) => {
+          console.warn('Session errored - received errorCode:', errorCode);
+      };
+
+      const loadYoti = async () => {
+          const { Yoti } = window
+          if (Yoti) {
+            console.info('Waiting for Yoti...')
+            await Yoti.ready()
+            console.info('Yoti is now ready')
+          } else {
+            console.error('Yoti client was not found!')
+          }
+      }
+
+      const createYotiWebShare = async () => {
+          const { Yoti } = window
+          if (Yoti) {
+            await Yoti.createWebShare({
+              name: 'share example basic attributes',
+              domId: 'yoti-share-button-basic-attributes',
+              sdkId: "<%= yotiClientSdkId %>",
+              hooks: {
+                sessionIdResolver,
+                completionHandler,
+                errorListener,
```

Comment:
Both `sessionIdResolver` and `errorListener` triggered on Yoti modal click. Unsure why this happens. Check the browser developer tools console. Should see something like this:
```
sessionIdResolver found id ss.v2.ChZQOUhTWlFCZFNaVzQ5VnFkYVJXVlFnEgkzLmxkNS5nYnI=
Session errored - received errorCode: undefined
```

Reply:
We will get rid of the error raised with no `errorCode` soon, but also that `errorListener` might be removed in next version, so I added a comment in the code.